crates/bevy_hierarchy/src/components/mod.rs

```diff
@@ -1,5 +1,26 @@
 mod children;
 mod parent;
 
+use bevy_ecs::world::{EntityWorldMut, Mut};
+
 pub use children::Children;
 pub use parent::Parent;
+
+/// Private trait to consolidate unsafe calls to [`get_mut_assume_mutable`](EntityWorldMut::get_mut_assume_mutable).
+pub(crate) trait GetHierarchyComponentsMut {
+    fn get_parent_mut(&mut self) -> Option<Mut<Parent>>;
+    fn get_children_mut(&mut self) -> Option<Mut<Children>>;
+}
+
+#[expect(unsafe_code)]
+impl GetHierarchyComponentsMut for EntityWorldMut<'_> {
+    fn get_parent_mut(&mut self) -> Option<Mut<Parent>> {
+        // SAFETY: The Parent component has no invariants.
```

Comment:
The doc comments for `get_mut_assume_mutable()` give the safety invariant as `` `T` must be a mutable component ``, which this safety comment doesn't satisfy.  If this is intended to be legal, then I think we need to update the documentation for that method.  

I thought the plan was that immutable components could only be updated by re-inserting them, though, and that this was not intended to be legal.

Reply:
I think updating the safety requirement for the `*_assume_mutable` methods is appropriate here in the interim. The ideal solution would be to use swapping to actually preserve immutability and avoid the unsafe methods, but that's a little cumbersome at the moment.

Reply:
"swapping" as in temporarily replacing `Children`/`Parent` with a dummy value so you can just have ownership over the component while you're editing it. Basically `core::mem::swap`. As an alternative, #16668 will add a `with_component` method which does this for you without the temporary value by directly triggering hooks/observers.

Reply:
I'd like to merge #16668 and use it here. This safety comment really confused me.